dvc/repo/params/show.py

```diff
@@ -103,14 +119,26 @@ def _collect_vars(repo, params) -> Dict:
 
 
 @locked
-def show(repo, revs=None, targets=None, deps=False, onerror: Callable = None):
+def show(
+    repo,
+    revs=None,
+    targets=None,
+    deps=False,
+    onerror: Callable = None,
+    stages=None,
```

Comment:
Does `stages` here accept stage at a custom path? It appears it doesn't, unless I'm missing something.

Reply:
> why do we need targets and stages?

Because they are not mutually exclusive. They have to be decoupled to support several scenarios, for example:

- A `stage` (stage_A) uses params from different `targets` (target_A, target_B).

`params_show(target_A, stage=stage_A)`

- Different parts of a single `target` (target_A) can be used for multiple `stages` (stage_A, stage_B).

`params_show(target_A, stage=stage_B)`

> But what if you have stages with same names?

I am not sure I follow. The `targets` and `stages` filters are applied separately, so it would be up to the user to provide unambiguous arguments.

> Does stages here accept stage at a custom path?

What's is a stage at a custom path?

Reply:
> What's is a stage at a custom path?

Like `path/to/dvc.yaml:mystage`. Because `mystage` could be defined in multiple dvc.yaml at different levels.